### DISCUSSION TITLE
Payload as String

### DISCUSSION THREAD

**Discussion by mawhtin** [panva/jose#364] [Q&A] [answered]
Hi Panva,

Any chance of supporting a String type payload? Currently it has to be a JSON object.

I looked at the specs, and it doesn't explicitly demand a JSON object.

jsonwebtoken library by Auth0 supports String type payloads, but I prefer jose.

If you're wondering, our use case needs the JWS as short as possible. Every byte counts.

**Comment by panva** [comment, marked as answer]
> I looked at the specs, and it doesn't explicitly demand a JSON object.

Sure does.

> JSON Web Token (JWT)
> A string representing a set of claims as a JSON object that is
> encoded in a JWS or JWE

> JWT Claims Set
> A JSON object that contains the claims conveyed by the JWT.

---

> Any chance of supporting a String type payload? Currently it has to be a JSON object.

What about the other existing APIs in `jose` for JWS?

- [Class: CompactSign](https://github.com/panva/jose/blob/v4.5.0/docs/classes/jws_compact_sign.CompactSign.md#readme)
- [Class: FlattenedSign](https://github.com/panva/jose/blob/v4.5.0/docs/classes/jws_flattened_sign.FlattenedSign.md#readme)

**Reply by mawhtin** [reply]
Thanks, appreciate it!
